packages/notifications/push/amplify_push_notifications_pinpoint/test/pinpoint_provider_test.dart

```diff
@@ -0,0 +1,213 @@
+// SPDX-License-Identifier: Apache-2.0
+
+// ignore: implementation_imports
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/analytics_client.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/endpoint_client/endpoint_client.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/event_client.dart';
+import 'package:amplify_flutter/amplify_flutter.dart';
+import 'package:amplify_push_notifications_pinpoint/src/pinpoint_provider.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+
+import 'pinpoint_provider_test.mocks.dart';
+import 'test_data/fake_notification_messges.dart';
+
+@GenerateMocks(
+  [
+    AmplifyAuthProviderRepository,
+    AWSIamAmplifyAuthProvider,
+    UserProfile,
+    AnalyticsClient,
+    EndpointClient,
+    EventClient,
+  ],
+)
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+  final pinpointProvider = PinpointProvider();
+  final mockAmplifyAuthProviderRepository = MockAmplifyAuthProviderRepository();
+  const notificationsPinpointConfig = NotificationsPinpointPluginConfig(
+    appId: 'APP_ID',
+    region: 'REGION',
+  );
+  final awsIamAmplifyAuthProvider = MockAWSIamAmplifyAuthProvider();
+  final mockAnalyticsClient = MockAnalyticsClient();
+  group('PinpointProvider', () {
+    test('init fails when retreiving an Auth provider was not successfull', () {
+      when(mockAmplifyAuthProviderRepository.getAuthProvider(any))
+          .thenReturn(null);
+      expect(
+        () async => pinpointProvider.init(
+          config: notificationsPinpointConfig,
+          authProviderRepo: mockAmplifyAuthProviderRepository,
+        ),
+        throwsA(
+          isA<ConfigurationError>().having(
+            (e) => e.message,
+            'config error',
+            contains('No AWSIamAmplifyAuthProvider available'),
+          ),
+        ),
+      );
+    });
+
+    test('identifyUser fails when the Pinpoint Provider is not initialized',
+        () async {
+      expect(
+        () async => pinpointProvider.identifyUser(
+          userId: 'userId',
+          userProfile: MockUserProfile(),
+        ),
+        throwsA(
+          isA<ConfigurationError>().having(
+            (e) => e.message,
+            'Not configured',
+            contains('Provider is not initialized'),
+          ),
+        ),
+      );
+    });
+
+    test(
+        'constructEventInfo should return journey data when there is journey details in the payload',
+        () async {
+      final res = pinpointProvider.constructEventInfo(
+        notification: PushNotificationMessage.fromJson(androidJourneyMessage),
+      );
+      final properties = res.properties;
+      final source = res.source;
+      expect(properties.attributes.containsKey('journey_id'), true);
+      expect(source, PinpointEventSource.journey.name);
```

Comment:
nit: could use matchers throughout
```suggestion
      expect(source, equals(PinpointEventSource.journey.name));
```